platform/source/mbed_retarget.cpp

```diff
@@ -92,11 +92,11 @@ asm(" .global __use_full_stdio\n");
 using namespace mbed;
 
 #if defined(__MICROLIB)
-// Before version 5.03, we were using a patched version of microlib with proper names
-extern const char __stdin_name[]  = ":tt";
-extern const char __stdout_name[] = ":tt";
-extern const char __stderr_name[] = ":tt";
-
+// Microlib currently does not allow re-defining the pathnames for the
+// standard I/O device handles (STDIN, STDOUT, and STDERR).
+// It uses the default pathname ":tt" at library initialization to identify
+// them all.
+extern const char default_pathname[]  = ":tt";
```

Comment:
This is an improvement on the only microlib code, cos there's now only 1 copy, but it doesn't need to be public. Make it `static` instead of `extern`, or even just lose it altogether, and do `strcmp(":tt")`.

The name `default_pathname` is a bit off - it's more `console_pathname`.

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/4ccc802012c03bbd86d41d2c5369fd89960b0888..e7b05d500deef723bd97cb47ce7b715d55799a43).